libvips/foreign/jxlsave.c

```diff
@@ -479,13 +480,13 @@ vips_foreign_save_jxl_set_header(VipsForeignSaveJxl *jxl, VipsImage *in)
 
 	switch (in->BandFmt) {
 	case VIPS_FORMAT_UCHAR:
-		jxl->info.bits_per_sample = 8;
+		jxl->info.bits_per_sample = VIPS_CLIP(1, jxl->bitdepth, 8);
```

Comment:
`jxl->bitdepth` is already clipped to a minimum of 1.
```suggestion
		jxl->info.bits_per_sample = VIPS_MIN(jxl->bitdepth, 8);
```